flink-runtime/src/main/java/org/apache/flink/runtime/security/modules/HadoopModule.java

```diff
@@ -70,10 +71,22 @@ public void install() throws SecurityInstallException {
 
         try {
             KerberosLoginProvider kerberosLoginProvider = new KerberosLoginProvider(securityConfig);
-            if (kerberosLoginProvider.isLoginPossible()) {
-                kerberosLoginProvider.doLogin();
+            if (kerberosLoginProvider.isLoginPossible(true)) {
+                kerberosLoginProvider.doLogin(true);
                 loginUser = UserGroupInformation.getLoginUser();
 
+                if (HadoopUserUtils.isProxyUser((loginUser))
```

Comment:
I think it's good to add this in general not to change the original behavior. I'm just writing down a possible future consideration so no change needed for now.

This pointed me to an edge case where a user wants to use proxy user support w/ non-Hadoop delegation tokens like S3. The original implementation is not allowing to do this but maybe there is a need for it. I need some more time to consider this in-depth. Later on we might remove this check when the use-cases expecting that.

Reply:
Just wanted to confirm that we no need any change right now? Rather you comment is more for future, handling proxy user support w/ non-hadoop DT like S3. I thought about this case as well. Good point!